### PR TITLE
Officially deprecated VariableSizeScatterPlot

### PR DESCRIPTION
**Pull request by aaronayres35**
Part of #580 

`VariableSizeScatterPlot` has been commented as being deprecated, but no warning was raised.  This PR officially deprecates it by adding an explicit warning when importing it.

It also adds a trivial test of the deprecation warning which may or may not be overkill...

### CODE REVIEW

**Comment by rahulporuri** [comment]
Same here. I don't think there's a point to deprecating it officially now and removing it in a later release. I don't see any meaningful users of `VariableSizeScatterPlot` on GitHub.

File: examples/demo/advanced/cmap_variable_sized_scatter.py [hunk 1]
```diff
@@ -1,6 +1,5 @@
 """
 Draws a scatterplot of a set of random points of variable size and color.
- - This uses the non-standard renderer, VariableSizeScatterPlot
```
Comment (rahulporuri) [inline examples/demo/advanced/cmap_variable_sized_scatter.py:3]: looks like we should rename this file - it doesn't use `VariableSizeScatterPlot` anymore - just the `ColormappedScatterPlot`

**Review by rahulporuri — APPROVED** [review]
LGTM with one minor suggestion

File: examples/demo/advanced/cmap_variable_sized_scatter.py (hunk 1 above)
Comment (aaronayres35) [reply examples/demo/advanced/cmap_variable_sized_scatter.py:3]: It is still an example of a scatter plot that uses variable sized markers though.  It just uses this functionality on the `ScatterMap` class directly

I'm not sure what we would rename it to

Comment (rahulporuri) [reply examples/demo/advanced/cmap_variable_sized_scatter.py:3]: fair enough. the name conveys behavior, not which plot type it is. never mind. let's leave it as is.

**Pull request merged**